content/docs/faq-state.md

```diff
@@ -1,106 +1,106 @@
+[`props`](/docs/components-and-props.html)-ը («properties»-ի կրճատ տարբերակը) և [`state`](/docs/state-and-lifecycle.html)-ը պարզ օբյեկտներ են։ Չնայած երկուսն էլ պահում են ինֆորմացիա\` որը ազդում է արտապատկերման վրա, դրանք ունեն մեկ կարևոր տարբերություն. `props`-ը *փոխանցվում է* կոմպոնենտին (ինչպես ֆունկցիայի պարամետրերը),  մինչդեռ `state`-ը *տեղակայված է* կոմպոնենտի ներսում (ինչպես փոփոխականները հայտարարվում են ֆունկցիայի ներսում)։
 
-Here are some good resources for further reading on when to use `props` vs `state`:
-* [Props vs State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)
-* [ReactJS: Props vs. State](https://lucybain.com/blog/2016/react-state-vs-pros/)
+Ահա մի քանի օգտակար ռեսուրսներ լրացուցիչ ընթերցման համար, թե երբ օգտագործել `props` և երբ `state`։
+* [Props, թե State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)<sub>eng</sub>
+* [ReactJS: Props, թե State](https://lucybain.com/blog/2016/react-state-vs-pros/)<sub>eng</sub>
 
-### Why is `setState` giving me the wrong value? {#why-is-setstate-giving-me-the-wrong-value}
+### Ի՞նչու է `setState`-ը վերադարձնում սխալ արժեք {#why-is-setstate-giving-me-the-wrong-value}
 
-In React, both `this.props` and `this.state` represent the *rendered* values, i.e. what's currently on the screen.
+React-ում `this.props`-ը և `this.state`-ը ներկայացնում են *արտապատկերված* արժեքներ, այսինքն\` այն ինչ տվյալ պահին երևում է էկրանին։
 
-Calls to `setState` are asynchronous - don't rely on `this.state` to reflect the new value immediately after calling `setState`. Pass an updater function instead of an object if you need to compute values based on the current state (see below for details).
+`setState`-ի կանչերը անսինխրոն են - `setState` կանչելուց հետո, երբ ուզում եք անմիջապես արտացոլել նոր արժեքը\` հույս մի դրեք `this.state`-ի վրա։ Եթե ուզում եք հաշվարկել արժեքները հիմնվելով ընթացիք state-ի վրա, օբյեկտի փոխարեն `setState`-ին փոխանցեք թարմացնող ֆունկցիա։ (ստորև ներկայացված է մանրամասն օրինակ)։
 
-Example of code that will *not* behave as expected:
+Կոդի օրինակ\` որը *չի* աշխատի այնպես, ինչպես ակնկալվում է.
 
 ```jsx
 incrementCount() {
-  // Note: this will *not* work as intended.
+  // Նշում. սա *չի* աշխատի նախատեսվածին պես։
   this.setState({count: this.state.count + 1});
 }
 
 handleSomething() {
-  // Let's say `this.state.count` starts at 0.
+  // Պատկերացնենք `this.state.count`-ը սկսվում է 0-ից։
   this.incrementCount();
   this.incrementCount();
   this.incrementCount();
-  // When React re-renders the component, `this.state.count` will be 1, but you expected 3.
+  // Երբ React-ը արտապատկերի կոմպոնենտը `this.state.count`-ի արժեքը կլինի 1, չնայած ակնկալում ենք 3:
```

Comment:
ակնկալում ենք -> նշանակում ա, ՄԵՆՔ ենք ակնկալու. գրած ա you։ Նաև expected ա, որը անցյալ ա